app/controllers/authentication_controller.rb

```diff
@@ -0,0 +1,79 @@
+class AuthenticationController < ApplicationController
+  before_action :check_if_enabled
+
+  MISSING_EMAIL_ERROR = 'Please enter your email address.'.freeze
+  INVALID_EMAIL_ERROR = 'This doesn’t look like a valid email address – check you’ve entered it correctly.'.freeze
+
+  def sign_in
+    @address = params[:address]
+  end
+
+  def request_sign_in_token
+    unless params[:address].present?
+      flash.now[:error] = MISSING_EMAIL_ERROR
+      return render :sign_in
+    end
+
+    @address = params.require(:address)
+
+    email_alert_api.create_auth_token(
+      address: @address,
+      destination: process_sign_in_token_path
+    )
+  rescue GdsApi::HTTPUnprocessableEntity
+    flash.now[:error] = INVALID_EMAIL_ERROR
+    return render :sign_in
+  end
+
+  def process_sign_in_token
+    subscriber_id, redirect = read_token(params.require(:token))
+
+    if subscriber_id.nil?
+      @sign_in_link = sign_in_path
+      return render :expired_link
+    end
+
+    authenticate_subscriber(subscriber_id)
+
+    destination = safe_redirect_destination(redirect) || '/email/manage'
+    redirect_to destination
+  end
+
+private
+
+  def check_if_enabled
+    return render plain: '404 not found', status: :not_found unless ENV['SUBSCRIPTION_MANAGEMENT_ENABLED']
+  end
+
+  def read_token(token)
+    payload, = JWT.decode(token, secret, true, algorithm: 'HS256')
```

Comment:
There's a comma after payload ?

Reply:
Yup, that's a non-intuitive way of saying that `JWT.decode` returns an array of values and we're only interested in the first one.

Reply:
Ahh okay!

Reply:
yeah probably easier to read if we do `payload, _ = JWT.decode` - do you find that easier @1pretz1 ? 

I think I wrote this originally, can be hard to remember what syntax is a bit deceptive.

Reply:
Yeah it was originally like that but the linter complained about it

Reply:
oh - bad linter!